iotile_ext_cloud/iotile/cloud/utilities.py

```diff
@@ -28,3 +30,35 @@ def device_slug_to_id(slug):
         return int(short, 16)
     except ValueError as exc:
         raise ArgumentError("Invalid device slug with non-numeric components", error_mesage=str(exc), slug=slug)
+
+def device_id_to_slug(id):
```

Comment:
There are two ways that we have to represent device ids:

1. As a normal integer (a python `int`)
2. As a slug, which is that same integer written as a string of hex digits with `-` between every 4 hex digits prepended with a `d--` prefix.

This function should take an integer id and convert it to a slug id.  So this input should be an integer.  We never deal with strings of hex digits directly.  It's either an integer or a slug.

As background.  CoreTools uses integers throughout because it was designed as a hardware interaction library that should be useable without cloud access and the hardware uses opaque integers as serial numbers to identify each POD or NODE uniquely.  However, whenever we interact with iotile cloud (which is only done by the iotile-ext-cloud in CoreTools) we need to convert our integer ID to a slug.

Reply:
(and back to an integer sometimes) which is the purpose of these two functions.

Reply:
Ok, will change accordingly !

Reply:
Only edit from your requested changes : I am using `long` instead of `int` so as not to overflow when we reach the last ids